### ISSUE TITLE
Make REMOTE_ADDR optional

### ISSUE THREAD

**Issue by mschoettle** [soynatan/django-easy-audit#263]
The `user_logged_in` signal handler fails during tests when calling `client.force_login(...)` because the request has no IP (`REMOTE_ADDR`).

```py
def user_logged_in(sender, request, user, **kwargs):
        try:
            with transaction.atomic(using=DATABASE_ALIAS):
                login_event = audit_logger.login({
                    'login_type': LoginEvent.LOGIN,
                    'username': getattr(user, user.USERNAME_FIELD),
                    'user_id': getattr(user, 'id', None),
>                   'remote_ip': request.META[REMOTE_ADDR_HEADER]
                })
E               KeyError: 'REMOTE_ADDR'

[snip]/easyaudit/signals/auth_signals.py:21: KeyError
```

Would it be ok to change this to `request.META.get(REMOTE_ADDR_HEADER)`? `remote_ip` is already nullable so from the `LoginEvent` model perspective it is already supported. And the request signal handler already supports this: 

https://github.com/soynatan/django-easy-audit/blob/bbdf834d671811043e7bab4aaeb39903ac21f982/easyaudit/signals/request_signals.py#L44

**Comment by mschoettle** [comment]
@jheld Let me know what you think please. Happy to create a PR for this.

The alternative would be to disable auth events to be watched in tests but it would be nice to stay as close as possible to production.

**Comment by jheld** [comment]
Should any extra conceptual weirdness be at play here by making this change, it's evading me at this time. Given the examples you've shown in the code where it certainly allows `None`, I concur with your approach. Yes, please make a PR :)

**Comment by 6abc** [comment]
# if we modify auth_signals.py 
# like this it will work, can u pls make these changes ?
from django.contrib.auth import signals, get_user_model
from django.db import transaction
from django.utils.module_loading import import_string

from easyaudit.middleware.easyaudit import get_current_request
from easyaudit.models import LoginEvent
from easyaudit.settings import REMOTE_ADDR_HEADER, WATCH_AUTH_EVENTS, LOGGING_BACKEND, \
    DATABASE_ALIAS
from easyaudit.utils import should_propagate_exceptions

audit_logger = import_string(LOGGING_BACKEND)()


def user_logged_in(sender, request, user, **kwargs):
    try:
        with transaction.atomic(using=DATABASE_ALIAS):
            login_event = audit_logger.login({
                'login_type': LoginEvent.LOGIN,
                'username': getattr(user, user.USERNAME_FIELD),
                'user_id': getattr(user, 'id', None),
                #'remote_ip': request.META[REMOTE_ADDR_HEADER]
                'remote_ip': request.META.get('HTTP_X_FORWARDED_FOR', request.META['REMOTE_ADDR']).split(',')[0],
            })
    except Exception:
        if should_propagate_exceptions():
            raise


def user_logged_out(sender, request, user, **kwargs):
    try:
        with transaction.atomic(using=DATABASE_ALIAS):
            login_event = audit_logger.login({
                'login_type': LoginEvent.LOGOUT,
                'username': getattr(user, user.USERNAME_FIELD),
                'user_id': getattr(user, 'id', None),
                #'remote_ip': request.META[REMOTE_ADDR_HEADER]
                'remote_ip': request.META.get('HTTP_X_FORWARDED_FOR', request.META['REMOTE_ADDR']).split(',')[0],
            })
    except Exception:
        if should_propagate_exceptions():
            raise


def user_login_failed(sender, credentials, **kwargs):
    try:
        with transaction.atomic(using=DATABASE_ALIAS):
            request = get_current_request() # request argument not available in django < 1.11
            user_model = get_user_model()
            login_event = audit_logger.login({
                'login_type': LoginEvent.FAILED,
                'username': credentials[user_model.USERNAME_FIELD],
                #'remote_ip': request.META[REMOTE_ADDR_HEADER]
                'remote_ip': request.META.get('HTTP_X_FORWARDED_FOR', request.META['REMOTE_ADDR']).split(',')[0],
            })
    except Exception:
        if should_propagate_exceptions():
            raise


if WATCH_AUTH_EVENTS:
    signals.user_logged_in.connect(user_logged_in, dispatch_uid='easy_audit_signals_logged_in')
    signals.user_logged_out.connect(user_logged_out, dispatch_uid='easy_audit_signals_logged_out')
    signals.user_login_failed.connect(user_login_failed, dispatch_uid='easy_audit_signals_login_failed')

![image](https://github.com/soynatan/django-easy-audit/assets/97246854/4d230209-638d-4951-b790-550256696e3d)
